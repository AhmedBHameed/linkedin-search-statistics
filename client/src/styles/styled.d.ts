import 'styled-components';

/**
 * For more information whey we need this file @see https://styled-components.com/docs/api#create-a-declarations-file
 */
declare module 'styled-components' {
  export interface DefaultTheme {
    colors: {
      blue5: string;
      blue50: string;
      blue90: string;
      red: string;
      highlightText: string;
    };
  }
}
