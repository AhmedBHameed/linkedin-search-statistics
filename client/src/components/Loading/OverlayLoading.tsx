import {Spin} from 'antd';
import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  z-index: 999;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.1);
`;

const OverlayLoading: React.FC = () => (
  <Container data-testid="loading-overlay">
    <Spin size="large" />
  </Container>
);

export default OverlayLoading;
