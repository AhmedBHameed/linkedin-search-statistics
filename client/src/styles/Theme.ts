/**
 * The number 50 is the scale of the color according to Figma
 * @see https://www.figma.com/file/WCipxn54SiYJSVpZ3uTUW5/BiomeDx-Styleguide-und-Report-%234?node-id=13%3A140
 */

const highlightText = '#ffc069';

const lightTheme = {
  red: 'red',
  blue5: '#e7f2fe',
  blue50: '#047CB3',
  blue90: '#011e2f',
  highlightText,
};

const darkTheme = {
  red: '#ff7777',
  blue5: '#e7f2fe',
  blue50: '#047CB3',
  blue90: '#011e2f',
  highlightText,
};

export type ThemeProps = typeof darkTheme;
export {lightTheme, darkTheme};
