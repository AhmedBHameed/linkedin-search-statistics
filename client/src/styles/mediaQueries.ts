export const sm = 576;
export const md = 768;
export const lg = 992;
export const xl = 1200;
export const xxl = 1600;

const devices = {
  smMediaQuery: `(min-width: ${sm}px)`,
  mdMediaQuery: `(min-width: ${md}px)`,
  lgMediaQuery: `(min-width: ${lg}px)`,
  xlMediaQuery: `(min-width: ${xl}px)`,
  xxlMediaQuery: `(min-width: ${xxl}px)`,
};

export default devices;
