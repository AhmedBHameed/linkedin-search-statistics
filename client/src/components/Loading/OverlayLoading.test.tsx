import {render} from '@testing-library/react';
import {ThemeProvider} from 'styled-components';
import {mockMediaQuery} from '../../setupTests';
import {md} from '../../styles/mediaQueries';
import {lightTheme} from '../../styles/Theme';
import OverlayLoading from './OverlayLoading';

function renderComponent() {
  mockMediaQuery(md, md);

  return render(
    <ThemeProvider theme={{colors: lightTheme}}>
      <OverlayLoading />
    </ThemeProvider>
  );
}

describe('Loading', () => {
  it('matches snapshot', async () => {
    const {container} = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
