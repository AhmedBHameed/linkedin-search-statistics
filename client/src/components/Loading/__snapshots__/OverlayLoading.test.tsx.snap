// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading matches snapshot 1`] = `
<div>
  <div
    class="sc-bdnxRM gWCkoE"
    data-testid="loading-overlay"
  >
    <div
      class="ant-spin ant-spin-lg ant-spin-spinning"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>
</div>
`;
